core/envelope-sim/src/main/kotlin/fr/sncf/osrd/envelope_sim/etcs/ETCSBrakingCurves.kt

```diff
@@ -40,220 +45,213 @@ enum class BrakingType {
+    for (limitOfAuthority in sortedLimitsOfAuthority) {
+        res.add(computeLOABrakingCurves(limitOfAuthority, context, envelope, 0.0))
     }
-    val indCurveEoa = Envelope.make(*indCurveEoaParts.toList().toTypedArray())
-    val indCurveSvl = Envelope.make(*indCurveSvlParts.toList().toTypedArray())
-
-    // The indication curve is the minimum of the SvL curve maintaining release speed and the EoA
-    // curve, between [minBeginPos, targetPosition]. We need the 2 envelopes to have the exact same
-    // position-range.
-    return minEnvelopes(indCurveEoa, indCurveSvl)
+    return res
 }
 
-/** Compute braking curves at every limit of authority. */
-fun addBrakingCurvesAtLOAs(
+/**
+ * Compute braking curves at every end of authority until their speed intersects with inputted
+ * envelope.
+ */
+fun computeBrakingCurvesAtEOAs(
     envelope: Envelope,
     context: EnvelopeSimContext,
-    limitsOfAuthority: Collection<LimitOfAuthority>
-): Envelope {
-    val sortedLimitsOfAuthority = limitsOfAuthority.sortedBy { it.offset }
-    val beginPos = envelope.beginPos
-    val maxSpeedEnvelope = envelope.maxSpeed
-    var envelopeWithLoaBrakingCurves = envelope
-    var builder = OverlayEnvelopeBuilder.forward(envelopeWithLoaBrakingCurves)
+    endsOfAuthority: Collection<EndOfAuthority>
+): List<Map<BrakingCurveType, ETCSBrakingCurve?>> {
+    val sortedEndsOfAuthority = endsOfAuthority.sortedBy { it.offsetEOA }
+    val res = mutableListOf<Map<BrakingCurveType, ETCSBrakingCurve?>>()
+    for (endOfAuthority in sortedEndsOfAuthority) {
+        res.add(computeEOABrakingCurves(endOfAuthority, context, envelope, 0.0))
+    }
+    return res
+}
 
-    for (limitOfAuthority in sortedLimitsOfAuthority) {
-        val targetPosition = limitOfAuthority.offset.distance.meters
-        assert(targetPosition > 0.0)
-        val targetSpeed = limitOfAuthority.speed
-        assert(targetSpeed > 0.0)
-        val fullIndicationCurve =
-            computeEbdFullIndicationCurve(context, targetPosition, targetSpeed, maxSpeedEnvelope)
-        val endOfIndicationCurve = fullIndicationCurve.endPos
-        assert(endOfIndicationCurve <= targetPosition)
-
-        val fullIndCurveWithMaintain: Envelope
-        if (endOfIndicationCurve < targetPosition) {
-            // Maintain target speed until target position, i.e. LOA.
-            val maintainTargetSpeedCurve =
-                EnvelopePart.generateTimes(
-                    listOf(EnvelopeProfile.CONSTANT_SPEED),
-                    doubleArrayOf(endOfIndicationCurve, targetPosition),
-                    doubleArrayOf(targetSpeed, targetSpeed)
-                )
-            fullIndCurveWithMaintain = Envelope.make(fullIndicationCurve, maintainTargetSpeedCurve)
-        } else {
-            fullIndCurveWithMaintain = Envelope.make(fullIndicationCurve)
-        }
+/** Compute LoA braking curves: compute EBD-based curves for LoA. */
+private fun computeLOABrakingCurves(
```

Comment:
To have a more distinct naming:
```suggestion
private fun computeBrakingCurvesAtOneLOA(
```

Reply:
Fixed in https://github.com/OpenRailAssociation/osrd/pull/11616/commits/98ef587440bd3fc6f02a6445af2fad1f839415f8